### PR TITLE
xgo: Use statusim/xgo Docker image, based on Go 1.9.2 instead of 1.8.3

### LINKED ISSUE #391: Make statusim/xgo image out of farazdagi/xgo
# Problem

In development, we use Go 1.9.x branch but a docker image that we use to build mobile binaries uses Go 1.7.1.

# Implementation

We should upgrade the image to the most recent Go 1.9.x version.

# Acceptance Criteria

1. The built binaries work correctly when compiled in the app.
2. `statusim/xgo` is deployed to our dockerhub.
3. `statusim/xgo` is used for cross-compilation instead of `farazdagi/xgo`.

# Notes

#337 should be done first.

go-ethereum does not use [xgo](https://github.com/karalabe/xgo) for mobile builds anymore but [gomobile](https://github.com/golang/mobile).

# Future Steps

Decide if want to continue using `xgo` instead of `gomobile`.

### PR DESCRIPTION
**Pull request by pedropombeiro**
This PR replaces the `farazdagi/xgo` Docker image based on xgo 1.8.3 with a status-im image based on xgo 1.9.2. It changes the xgo Dockerfiles and the Makefile. I've decided to reuse the versioning system from the base xgo Docker image, which allows us to reference older images in the future in case something is broken (or we want to build a branch which uses an older xgo version).

One question, the GH issue suggests using `statusim/xgo` as the image name. Should it instead be `status-im/xgo`, in keeping with the GitHub organization name?

Remaining work from Ops:
- [ ] Update version reference at the bottom of https://github.com/status-im/status-go/wiki/Building to 1.9.*
- [ ] Optional: Set up an automated build in Docker Hub as described in https://github.com/status-im/status-go/issues/391#issuecomment-353801680.
- [ ] Optional: Upload `iPhoneSimulator9.3.sdk.tar.gz` to a CDN address controlled by Status (https://github.com/status-im/status-go/issues/391#issuecomment-353740549).

Once these bullet points have been addressed, the PR can be tested.

Closes #391

### CODE REVIEW

File: Makefile
Comment (PombeirP) [inline Makefile]: Any reason why we're not referencing the xgo executable with `$(GOBIN)/xgo` instead of `$(GOPATH)/bin/xgo`?

**Comment by adambabik** [comment]
Thanks for this PR. It looks gorgeous! 

> I've decided to reuse the versioning system from the base xgo Docker image, which allows us to reference older images in the future in case something is broken

Do I understand it correctly that when we decide to switch to e.g. Go 1.10 we would need to create a new repo in Docker Hub called `statusim/xgo-1.10.0` and probably disable automatic builds for `statusim/xgo-1.9.2` to not override it? I don't see any advantages of using this pattern over tags... Also, we probably won't need automated builds overall as this image changes very rarely.

> One question, the GH issue suggests using statusim/xgo as the image name. Should it instead be status-im/xgo, in keeping with the GitHub organization name?

`status-im/xgo` is probably better. I am still waiting for account confirmation regarding Docker Hub. We may already have one so we will reuse it.

**Comment by pedropombeiro** [comment]
Thanks @adambabik !

Regarding the image versioning, I agree that tags are better for our needs than the system that was chosen by the `xgo` image maintainer, I just defaulted to maintaining a 1-1 relationship between the image names.

So I'll do a couple of things:
- [x] change the versioning of `status-im/xgo` to be based on tags
- [x] change scripts to reference `status-im` entity instead of `statusim`

**Comment by pedropombeiro** [comment]
@adambabik what do you think of adding a Makefile target to build the xgo Docker images, as in your [PR](https://github.com/status-im/status-go/pull/521)? Just let me know and I'll take care of it.

**Comment by adambabik** [comment]
@PombeirP Great idea! We try to keep each build command in there so definitely 👍

**Review by adambabik — APPROVED** [review]

File: Makefile
Comment (adambabik) [inline Makefile]: We need to change this to `statusteam` as `-` is not allowed in a Docker ID. We have already `statusteam` organization created in Docker Cloud.

Comment (pedropombeiro) [reply Makefile]: @adambabik Done!

Comment (adambabik) [reply Makefile]: Thanks!

**Review by JekaMas — APPROVED** [review]

**Comment by pedropombeiro** [comment]
@JekaMas thanks for the review. Fixed the `Makefile` merge conflict with latest `develop`.

**Comment by pedropombeiro** [comment]
Thanks @adambabik. Would be cool if we could get the images built and hosted up on the `statusteam` organization, so that the local script works without modifications. Any idea if anyone is working on that?

**Comment by adambabik** [comment]
@PombeirP I am working on that. Just pushed `xgo` image. Trying to do the same with `xgo-ios-simulator` now.

**Pull request merged**
